Source/Details/_ASDisplayViewAccessiblity.mm

```diff
@@ -109,64 +121,112 @@ - (CGRect)accessibilityFrame
+/// Collect all subnodes for the given node by walking down the subnode tree and calculates the screen coordinates based on the containerNode and container. This is necessary for layer backed nodes or rasterrized subtrees as no UIView instance for this node exists.
+static void CollectAccessibilityElementsForLayerBackedOrRasterizedNode(ASDisplayNode *node, ASDisplayNode *containerNode, id container, NSMutableArray *elements)
 {
   ASDisplayNodeCAssertNotNil(elements, @"Should pass in a NSMutableArray");
 
+  // Iterate any node in the tree and either collect nodes that are accessibility elements
+  // or leaf nodes that are accessibility containers
   ASDisplayNodePerformBlockOnEveryNodeBFS(node, ^(ASDisplayNode * _Nonnull currentNode) {
-    // For every subnode that is layer backed or it's supernode has subtree rasterization enabled
-    // we have to create a UIAccessibilityElement as no view for this node exists
-    if (currentNode != containerNode && currentNode.isAccessibilityElement) {
-      UIAccessibilityElement *accessibilityElement = [ASAccessibilityElement accessibilityElementWithContainer:container node:currentNode];
-      [elements addObject:accessibilityElement];
+    if (currentNode != containerNode) {
+      if (currentNode.isAccessibilityElement) {
+        // For every subnode that is an accessibility element and is layer backed
+        // or it's supernode has subtree rasterization enabled, create a
+        // UIAccessibilityElement as no view for this node exists
+        UIAccessibilityElement *accessibilityElement = [ASAccessibilityElement accessibilityElementWithContainerView:container node:currentNode];
+        [elements addObject:accessibilityElement];
+      } else if (ASActivateExperimentalFeature(ASExperimentalTextNode2A11YContainer) &&
+                 ASIsLeafNode(currentNode) &&
+                 currentNode.accessibilityElementCount > 0) {
+        // In leaf nodes that are layer backed and acting as UIAccessibilityContainer
+        // (isAccessibilityElement == NO we call through to the
+        // accessibilityElements to collect all accessibility elements of this node
+        [elements addObjectsFromArray:currentNode.accessibilityElements];
+      }
     }
   });
 }
 
-static void CollectAccessibilityElementsForContainer(ASDisplayNode *container, UIView *view,
-                                                     NSMutableArray *elements) {
-  ASDisplayNodeCAssertNotNil(view, @"Passed in view should not be nil");
-  if (view == nil) {
+///// Called from CollectAccessibilityElements for nodes that are returning YES for isAccessibilityContainer to collect all subnodes accessibility labels as well as custom actions for nodes that have interactive accessibility traits enabled. Furthermore for ASTextNode's it also aggregates all links within the attributedString as custom action
```

Comment:
Split this comment, rename function `AggregateSubtreeAccessibilityLabelsAndCustomActions`. "Or" doesn't seem right

Reply:
Done